spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,295 @@
+# Type references and ownership
+
+## Related reading
+1. [Kubernetes garbage collection](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)
+
+## Goals
+1. Provide a way for customers to express relationships between Azure resources in Kubernetes in an idiomatic way.
+2. Provide automatic ownership and garbage collection (using [Kubernetes garbage collection](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)) 
+   where appropriate (e.g. ResourceGroup as an owner of all the resources inside of it)
+   1. Ideally ResourceGroup is handled the same as other owners and isn't special cased. 
+3. Define how Kubernetes interacts with Azure resources not created/managed by Kubernetes, for example resources which were created prior to the 
+   customer onboarding to the Azure Service Operator.
+ 
+## Non-goals
+??
+
+## Different kinds of resource relationships in Azure
+#### Related/linked resources 
+Two resources are related to one another ("has-a" relationship), but there is no ownership.
+Example: [VMSS -> Subnet](https://github.com/Azure/k8s-infra/blob/master/apis/microsoft.compute/v1/virtualmachinescaleset_types.go#L169) ([json schema](https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.json))
+
+#### Owner and dependent
+Two resources have a relationship where one is owned by the other.
+
+Examples:
+1. [RouteTable -> Route](https://github.com/Azure/k8s-infra/blob/master/apis/microsoft.network/v1/routetable_types.go#L18) ([json schema](https://schema.management.azure.com/schemas/2020-03-01/Microsoft.Network.json))
+2. BatchAccount -> Pool ([json schema](https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Batch.json))
+3. ResourceGroup -> Any resource
+
+This relationship tells us two things:
+1. Where to create/manage the dependent resource (this `Route` goes in this particular `RouteTable`, this `RouteTable` has this `Route`)
+2. That the dependent resource should be deleted when the parent resource is deleted. TODO: Race conditions here
+
+Note that sometimes an owning type has its dependent type embedded as a property 
+(for example: [RouteTable has a property routes - look for "RouteTablePropertiesFormat"](https://schema.management.azure.com/schemas/2020-03-01/Microsoft.Network.json)
+Most types don't embed the dependent resource directly in the owning resource. We need to cater to both scenarios.
+
+## What does expressing these relationships look like in existing solutions?
+
+### Related/Linked resources
+
+#### What ARM does
+These are just properties (often but not always called `id`) which refer to the fully qualified ARM ID of another resource. For example see a
+[sample deployment template for a VMSS refering to an existing vnet](https://github.com/Azure/azure-quickstart-templates/blob/master/201-vmss-existing-vnet/azuredeploy.json#L136).
```

Comment:
Suggest copying a fragment of that example in here so that readers don't have to look elsewhere for details.